apps/site/assets/ts/shuttles/components/ShuttlesMap.tsx

```diff
@@ -1,46 +1,103 @@
+  MapData,
+  MapMarker,
+  Polyline,
+  IconOpts
+} from "../../leaflet/components/__mapdata";
 
 interface Props {
   shapes: Shape[];
   stops: Stop[];
   center?: Coordinates;
 }
 
+/* eslint-disable @typescript-eslint/camelcase */
+const makeMarkers = (stops: Stop[]): MapMarker[] =>
+  stops.map((stop: Stop, index) => {
+    let icon: string;
+    switch (stop.type) {
+      case "shuttle":
+        icon = "mode-bus-default";
+        break;
+      case "rail_affected":
+        icon = "service-none-default";
+        break;
+      default:
+        icon = "circle-t-small";
+        break;
+    }
+
+    const icon_opts: IconOpts = {
+      icon_size: [16, 16],
+      icon_anchor: [8, 8]
+    };
+    const id = `${stop.name}-${index}`;
+    const { longitude, latitude } = stop;
+    const rotation_angle = 0;
+    const tooltip = null;
+
+    return {
+      icon,
+      icon_opts,
+      id,
+      latitude,
+      longitude,
+      rotation_angle,
+      tooltip,
+      shape_id: stop.id
+    };
+  });
+
+const makePolylines = (shapes: Shape[]): Polyline[] =>
+  shapes.map((shape: Shape, index) => {
+    const color = shape.is_shuttle_route ? "black" : "blue";
+    return {
+      color,
+      "dotted?": shape.is_shuttle_route,
+      id: `${shape.id}-${index}`,
+      positions: shape.polyline,
+      weight: 8
+    };
+  });
+
 const ShuttlesMap = ({
   shapes,
   stops,
   center
-}: Props): ReactElement<HTMLElement> => (
-  <div
-    className="m-shuttles-map"
-    style={{
-      border: "1px dotted gray",
-      backgroundColor: "#ee000011",
-      height: "444px",
-      fontSize: "12px"
-    }}
-  >
-    Center?: {JSON.stringify(center)}
-    <details>
-      <summary>Shapes: ({shapes.length})</summary>
-      {shapes.map(shape => (
-        <p key={shape.id}>
-          {shape.id} ({shape.is_shuttle_route ? "shuttle" : "not shuttle"}):{" "}
-          {shape.direction_id}
-        </p>
-      ))}
-    </details>
-    <details>
-      <summary>Stops: ({stops.length})</summary>
-      {stops.map(stop => (
-        <p key={stop.id}>
-          {stop.id} {stop.name} ({stop.type}) - ({stop.latitude},
-          {stop.longitude})
-        </p>
-      ))}
-    </details>
-  </div>
-);
+}: Props): ReactElement<HTMLElement> | null => {
+  if (typeof window !== "undefined") {
+    /* eslint-disable-next-line global-require */
+    const getBounds = require("../../leaflet/bounds").default;
+    const bounds = getBounds(
+      makeMarkers(stops.filter(stop => stop.type === "rail_affected"))
+    );
+    const mapCenter = center || {
+      longitude: -71.05891,
+      latitude: 42.360718
+    };
+
+    /* eslint-disable @typescript-eslint/camelcase */
+    const mapData: MapData = {
+      height: 400,
+      width: 400,
+      default_center: mapCenter,
+      markers: makeMarkers(stops),
+      stop_markers: makeMarkers(stops),
+      polylines: makePolylines(shapes),
+      tile_server_url: "https://mbta-map-tiles-dev.s3.amazonaws.com",
```

Comment:
This strikes me as something that should not be hard-coded, although annoyingly, it looks like this is only properly defined on the server (since with the other maps the server is responsible for everything).

As a quick solution, we could pass the URL into the component as a separate prop, and get the value from `Application.fetch_env!(:site, :tile_server_url)` in the server-rendered template.

Reply:
This has been tweaked to use `Application.fetch_env!(:site, :tile_server_url)`, thanks!